common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryCloud.java

```diff
@@ -111,4 +114,25 @@ void setIsValidated(final boolean isValidated) {
         mIsValidated = isValidated;
     }
 
+    /**
+     * Checks for equality with passed in AzureActiveDirectoryCloud object based on
+     * PreferredNetworkHostName and PreferredCacheHostName.
+     *
+     * @param cloudObjectToCheck AzureActiveDirectoryCloud object to check against.
+     * @return true if PreferredNetworkHostName and PreferredCacheHostName matches, otherwise false.
+     */
+    @Override
+    public boolean equals(@Nullable final Object cloudObjectToCheck) {
+        if(cloudObjectToCheck == null || getClass() != cloudObjectToCheck.getClass()) {
+            return false;
+        }
+
+        return  mPreferredNetworkHostName.equals(((AzureActiveDirectoryCloud) cloudObjectToCheck).mPreferredNetworkHostName) &&
+                mPreferredCacheHostName.equals(((AzureActiveDirectoryCloud) cloudObjectToCheck).mPreferredCacheHostName);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(mPreferredNetworkHostName, mPreferredCacheHostName);
+    }
```

Comment:
Can use Lombok's equalAndHashCode that allows to include and exclude specific fields from equals...any reason not to use it?

Reply:
Updated to use lombok.